enclave_build/docs/cache.md

```diff
@@ -0,0 +1,127 @@
+|________index.json        -> will point to the manifest blob for each image
+|________oci-layout
+|            blobs/sha256/
+|            |_______98806831de537d56f19b4b3cbf6ed80187374a1521a18d2d0c5f8a0e3962b2ae -> manifest_hash will point to config and layers
+|            |_______feb5d9fea6a5e9606aa995e879d862b825965ba48de054caab5ef356dc6b3412 -> config_hash
+|            |_______2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54 -> layer_hash
+```
+
+### Examples
+
+#### Index
+
+index.json
+```
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+      "size": 425,
+      "digest": "sha256:98806831de537d56f19b4b3cbf6ed80187374a1521a18d2d0c5f8a0e3962b2ae",
+      "annotations": {
+        "org.opencontainers.image.ref.name": "docker.io/test-oci:hello"
+      }
+    }
+  ]
+}
+```
+
+As we can see, the index holds one manifest entry, for one image stored. The number of manifests increases with each newly stored image. The manifest hash will point us to the manifest blob.
+
+#### Layout file
+
+oci-layout:
+```
+{"imageLayoutVersion":"1.0.0"}
+```
+
+Will simply define the OCI image layout version we are using.
+
+#### Manifest
+
+98806831de537d56f19b4b3cbf6ed80187374a1521a18d2d0c5f8a0e3962b2ae
+```
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+  "config": {
+    "mediaType": "application/vnd.docker.container.image.v1+json",
+    "digest": "sha256:feb5d9fea6a5e9606aa995e879d862b825965ba48de054caab5ef356dc6b3412",
+    "size": 1469
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+      "digest": "sha256:2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54",
+      "size": 2479
+    }
+  ]
+}
+```
+
+The manifest is our guide for a particular image. It tells us which components among the blobs belong to the image it defines. It will tell us the digest of the content for the config file and each layer. With this information we can search the blobs.
+
+#### Config
+
+The configuration file will define the behavior of the image inside the container, platform dependencies and interaction with the filesystem composed from the layers. Nitro CLI's usage of the cache should not concern with the contents of this file (excepting UTF-8 and digest validation).
+
+#### Layers
+
+Layers are serialized filesystem changes that go on top of each other, starting from the base filesystem we create the image with. The file will be binary content that, from Nitro CLI's point of view, just has to obey the content digest in the manifest.
+
+## Operations
+
+The operations offered by the `cache` module are as follows:
+
+### Create
+
+When building our first image, the cache will probably not exist. This operation will create, prior to storing an image, an empty cache. It will contain the cache root folder, an index file, a blobs folder and the oci-layout.
```

Comment:
The files in the empty cache will have a default, initial state, or are they going to be empty files? Please add details about this.